pkgs/by-name/es/esp-idf/riscv32-esp-elf.nix

```diff
@@ -0,0 +1,27 @@
+{
+  autoPatchelfHook,
+  darwin,
+  fetchurl,
+  lib,
+  sourceInfo ?
+    (builtins.fromJSON (builtins.readFile ./source-info.json)).tools.${stdenv.system}.riscv32-esp-elf,
+  stdenv,
+}:
+stdenv.mkDerivation {
+  inherit (sourceInfo) pname version;
+  src = fetchurl { inherit (sourceInfo) name url sha256; };
+
+  dontBuild = true;
+  installPhase = "cp -R . $out";
+
+  preFixup = lib.optionalString stdenv.isDarwin ''
+    for lib in $out/libexec/gcc/riscv32-esp-elf/${builtins.head (lib.splitString "_" sourceInfo.version)}/*.so
```

Comment:
Since you are dealing with pre built files here a lot, we need to add something like `sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];` into the `meta` section. Also make sure, that the `platform` in `meta` is properly set.